drivers/sensor/amd_sb_tsi/sb_tsi.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT amd_sb_tsi
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/logging/log.h>
+#include "sb_tsi.h"
+
+LOG_MODULE_REGISTER(AMD_SB_TSI, CONFIG_SENSOR_LOG_LEVEL);
+
+struct sb_tsi_data {
+	uint8_t sample_int;
+	uint8_t sample_dec;
+};
+
+struct sb_tsi_config {
+	struct i2c_dt_spec i2c;
+};
+
+static int sb_tsi_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct sb_tsi_data *data = dev->data;
+	const struct sb_tsi_config *config = dev->config;
+	enum pm_device_state pm_state;
+	int res;
+
+	(void)pm_device_state_get(dev, &pm_state);
+	if (pm_state != PM_DEVICE_STATE_ACTIVE) {
+		return -EIO;
+	}
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
```

Comment:
I think this is the wrong channel, die temperature is usually the internal temperature of the sensor used for calibration. I think you want `SENSOR_CHAN_AMBIENT_TEMP`

Reply:
Done